### DISCUSSION TITLE
JPEG buffered to TFT - is it possible? (to show images faster)

### DISCUSSION THREAD

**Discussion by acritum** [Bodmer/TFT_eSPI#3278] [Q&A - General]
I'm using the standard example from the library TJpg_Decoder to show jpg files from a SD card on a TFT screen (ST7789 320x240):
TJpgDec.setCallback(tft_output)
TJpgDec.drawSdJpg(0, 0, filename);

It works rather fast but still even small 320x240 files are shown not immediately but line-by line, and it doesn't look nice when the files are bigger. I wonder if there exists some work-around to load the jpg file into some kind of buffer, and when the jpg is finally loaded and decoded, copy this buffer into the video memory to make the TFT screen show the image instantly (or at least much faster then when it is loaded, decoded and shown at the same time). Maybe someone thought about this and found some solution?

**Comment by zachtos** [comment]
I am having the same issue, 150-200ms to display a 320x480ms jpeg. It makes my project unworkable with this library. Images load nice and fast when using sprites and internal memory. I'm not even sure if copying to sprite would matter for SD card?  Maybe we can put a header file on the card to remove the need for decoding somehow?
